Prepare EML for GBIF upload

sannegovaert opened the pull request:
This script prepares EML data extracted from [IMIS](https://www.vliz.be/en/imis?module=dataset) for upload to a [GBIF IPT](https://gbif.org/ipt).

sannegovaert committed:
Create prepare_eml_for_GBIF.Rmd

sannegovaert committed:
upload data

sannegovaert committed:
fix typo's

sannegovaert committed:
remove dx. in link

sannegovaert committed:
reorder

sannegovaert committed:
Clean Resource citation identifier

sannegovaert committed:
fix typo

sannegovaert committed:
remove taxonomicCoverage id

sannegovaert committed:
account for more than 1 species in taxonomic coverage

sannegovaert committed:
update description

sannegovaert committed:
update description

peterdesmet commented on data/2011_RIVIERPRIK/processed/eml.xml:
```diff
@@ -0,0 +1,242 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
I would organize the data similarly to bird-tracking:

    data/processed/DATASET_NAME/eml.xml

I would not store the raw eml, since we can get that one directly from IMIS.

peterdesmet committed:
Rename file

peterdesmet committed:
Minor textual updates (not to code)

peterdesmet committed:
Correct link in description

peterdesmet commented:
I also think it's best to remove the external link to IMIS (like `https://www.vliz.be/en/imis?dasid=5867`) as part of this script, since we have the DOI pointing to the same thing.

peterdesmet commented:
And I would remove all emails of all creators/contacts, since there are still some odd emails there. ORCID is sufficient.

peterdesmet committed:
Use shorter url

sannegovaert committed:
Merge branch 'main' into prepare_eml_for_gbif

sannegovaert committed:
change folder structure

sannegovaert committed:
read eml directly from IMIS url

sannegovaert committed:
set all mailAdresses to NULL, because of off emails

sannegovaert committed:
update processed eml

peterdesmet commented on data/processed/2011_RIVIERPRIK/eml.xml:
```diff
@@ -0,0 +1,332 @@
+      <para><![CDATA[This is an acoustic telemetry dataset published by the Research Institute for Nature and Forest (INBO). It contains animal (fish) tracking data collected by what later became the Permanent Belgian Acoustic Receiver Network <a href="https://lifewatch.be/en/fish-acoustic-receiver-network"> (https://lifewatch.be/en/fish-acoustic-receiver-network)</a> for the project/study 2011_r ...
+      <para>The disruption of longitudinal and lateral connectivity of rivers has led to ecological catastrophes such as the extinction of several diadromous fish species. River lamprey is an important indicator species for the integrity of ecosystems and connectivity within and between catchment areas. In the highly fragmented Scheldt river basin first restoration actions are undertaken, such as ...
+      <para>The study was commissioned by the Vlaamse Waterweg NV.</para>
+      <para>Data have been standardized to Darwin Core using the &lt;a href="https://inbo.github.io/etn/"&gt;etn&lt;/a&gt; package and are downsampled to the first detection per hour. The original data are managed in the European Tracking Network data platform (&lt;a href="https://lifewatch.be/etn/"&gt;https://lifewatch.be/etn/&lt;/a&gt;) and are available in Buysse et al. (2020, &lt;a href="http ...
```
It looks like this paragraph also needs to be wrapped in `<![CDATA[` (cf. the first one). Can you update the script to do that?

peterdesmet reviewed (changes requested):
Looks good. The last paragraph in the description (that is added with our `write_eml()` function) has escaped HTML characters. I think we should wrap that in `<![CDATA[`.

sannegovaert committed:
Update etn-occurrences.Rproj

sannegovaert committed:
wrap  more paragraphs in <![CDATA[

sannegovaert committed:
apply to all datasets

sannegovaert commented:
I updated the script so to wrap more paragraphs with `<![CDATA[`.
It seems that more metadata is added from IMIS. I do not understand why pubdate is going back in time however.

(I forgot about this issue)

peterdesmet reviewed (changes requested):
Last paragraph `&lt;![CDATA[` has escaped characters, rather than `<![CDATA[`. I don't think this will work in IPT import. 😬

sannegovaert committed:
reorder CDATA wrap code

sannegovaert committed:
update to EML 2.2.0

- ![CDATA[ ]]> wrappers are not needed anymore
- lists of paragraphs ara replaced by <p></p> tags

sannegovaert commented:
As we discussed and tested yesterday on the IPT:
- `<![CDATA[ ]]>` wrappers are not needed anymore
- lists of paragraphs can be replaced by `<p></p>` tags

sannegovaert committed:
split in paragraphs and add <p></p> tags

Sometimes the <p></p> tags are missing in the source eml. This solves that problem.

sannegovaert committed:
typo

peterdesmet commented:
@sannegovaert I think this PR can be closed? EML will be provided by IMIS.
